test/test_masked.py

```diff
@@ -162,10 +164,110 @@ def apply_masked_normalization_along_dim(op, input, *args, **kwargs):
+        return obj.to_dense()
+    return obj
+
+
+def to_strided(obj):
+    """Convert the tensor content of object to strided tensor content.
+    """
+    return torch.utils._pytree.tree_map(_tensor_to_strided, obj)
+
+
+def to_sparse_coo(obj):
+    """Convert the tensor content of object to sparse coo tensor content.
+    """
+    return torch.utils._pytree.tree_map(torch.Tensor.to_sparse, obj)
+
+
+def to_sparse_csr(obj):
+    """Convert the tensor content of object to sparse csr tensor content.
+    """
+    return torch.utils._pytree.tree_map(torch.Tensor.to_sparse_csr, obj)
+
+
+class mask_layouts(_TestParametrizer):
+    """Decorator class for parametrization of test function with an input
+    layout argument and an extra argument of sample inputs generator.
+    The sample_inputs generator provides samples with all supported
+    layouts for the mask argument.
+    """
+    def _parametrize_test(self, test, generic_cls, device_cls):
+
+        @wraps(test)
+        def wrap(self, layout, device, dtype, op):
+            layout_name = str(layout).lstrip('torch.')
+            if layout == torch.strided:
+                # strided layouts are always supported
+                sample_inputs_func = op.sample_inputs
+            elif layout == torch.sparse_coo:
+                if not op.supports_sparse:
+                    raise unittest.SkipTest(f"{op.name} does not support inputs with {layout_name} layout")
+                sample_inputs_func = op.sample_inputs_sparse_coo
+            elif layout == torch.sparse_csr:
+                if not op.supports_sparse_csr:
+                    raise unittest.SkipTest(f"{op.name} does not support inputs with {layout_name} layout")
+                sample_inputs_func = op.sample_inputs_sparse_csr
+            else:
+                raise NotImplementedError(f'{layout}')
+
+            def sample_inputs_generator():
+                for sample_input in sample_inputs_func(device, dtype):
+                    mask = sample_input.kwargs.get('mask')
+                    if mask is None:
+                        yield sample_input
+                    else:
+                        if layout == sample_input.input.layout:
+                            yield sample_input
+                        if layout != torch.strided:
+                            sample_input_kwargs = sample_input.kwargs.copy()
+                            sample_input_kwargs.update(mask=mask.to_dense())
+                            yield SampleInput(sample_input.input.clone(),
+                                              args=sample_input.args,
+                                              kwargs=sample_input_kwargs)
+                        if layout != torch.sparse_coo and op.supports_sparse:
+                            sample_input_kwargs = sample_input.kwargs.copy()
+                            if mask.layout == torch.sparse_csr:
+                                # TODO: remove this if-block when sparse csr supports to_sparse
```

Comment:
@pearu Could I ask you to send that PR separately as a follow up either as part of this stack or separately right after this lands?